app/components/post_with_channel_info/channel_info/channel_info.tsx

```diff
@@ -15,51 +18,86 @@ const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
     container: {
         flex: 1,
         flexDirection: 'row',
+        alignItems: 'center',
         marginVertical: 8,
-        paddingHorizontal: 16,
+        paddingHorizontal: 8,
+    },
+    channelContainer: {
+        borderRadius: 4,
+        overflow: 'hidden',
     },
     channel: {
         ...typography('Body', 75, 'SemiBold'),
-        color: theme.centerChannelColor,
-        marginRight: 5,
+        color: changeOpacity(theme.centerChannelColor, 0.72),
         flexShrink: 1,
+        paddingHorizontal: 8,
+        paddingVertical: 4,
     },
     teamContainer: {
         borderColor: theme.centerChannelColor,
-        borderLeftWidth: StyleSheet.hairlineWidth,
         flexShrink: 1,
     },
     team: {
-        ...typography('Body', 75, 'Light'),
-        color: theme.centerChannelColor,
-        marginLeft: 5,
+        ...typography('Body', 75, 'Regular'),
+        color: changeOpacity(theme.centerChannelColor, 0.72),
+        marginLeft: 8,
     },
 }));
 
 type Props = {
+    channelId: ChannelModel['id'];
     channelName: ChannelModel['displayName'];
     teamName: TeamModel['displayName'];
     testID?: string;
 }
 
-function ChannelInfo({channelName, teamName, testID}: Props) {
+function ChannelInfo({channelId, channelName, teamName, testID}: Props) {
     const theme = useTheme();
     const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+
+    const [isPressed, setIsPressed] = useState<Boolean>(false);
+
+    const channelNameStyle = useMemo(() => (
+        [styles.channel, isPressed ? {color: theme.buttonBg} : null]
+    ), [isPressed]);
+
+    const teamContainerStyle = useMemo(() => (
+        [styles.teamContainer, isPressed ? null : {borderLeftWidth: StyleSheet.hairlineWidth}]
+    ), [isPressed]);
+
+    const onChannelSwitch = useCallback(() => {
+        switchToChannelById(serverUrl, channelId);
+    }, [serverUrl]);
+
+    const togglePressed = useCallback(() => {
+        setIsPressed(!isPressed);
```

Comment:
To avoid the dependency and potential re-rendering, you can use the function form of set state.
```suggestion
        setIsPressed((v) => !v);
```